src/routes/profile.svelte

```diff
@@ -131,22 +138,17 @@
 
   let activeTab = tabs.timeline;
 
-  function changeUsername({ detail }) {
-    api.patch(`/accounts/${account.email}/telegram`, {
-      data: {
-        telegram_username: detail,
-      },
-    }).then((resp) => {
-      if (resp.ok) {
-        account.telegram_username = detail;
-      } else {
-        if (resp.status === 400) {
-          resp.json().then((error) => console.error(error.message));
-        } else {
-          resp.text().then(console.error);
-        }
-      }
-    });
+  async function changeUsername({ detail }) {
+    try {
+      await api.json(api.patch(`/accounts/${account.email}/telegram`, {
+        data: { telegram_username: detail },
+        csrfToken: account.csrf_token,
+      }));
+      account.telegram_username = detail;
+      $session.account = account;
```

Comment:
You are overriding the logged in user with the user whose profile is currently being viewed. I logged in as an admin, viewed a student's profile, changed their telegram, and all of a sudden I became the student!